packages/oasgraph/src/oas_3_tools.ts

```diff
@@ -194,36 +195,56 @@ export function getBaseUrl (
 
   // check for local servers
   if (Array.isArray(operation.servers) && operation.servers.length > 0) {
-    let url = buildUrl(operation.servers[0])
-
-    if (Array.isArray(operation.servers) && operation.servers.length > 1) {
-      logHttp(`Warning: Randomly selected first server ${url}`)
-    }
+    let url = buildUrl(getPreferedServer(operation.servers, preferedScheme))
 
     return url.replace(/\/$/, '')
   }
 
   if (Array.isArray(oas.servers) && oas.servers.length > 0) {
-    let url = buildUrl(oas.servers[0])
-
-    if (Array.isArray(oas.servers) && oas.servers.length > 1) {
-      logHttp(`Warning: Randomly selected first server ${url}`)
-    }
+    let url = buildUrl(getPreferedServer(oas.servers, preferedScheme))
 
     return url.replace(/\/$/, '')
   }
 
   throw new Error('Cannot find a server to call')
 }
 
+/**
+ * Return the first server that use the prefered scheme. If none match, return the first server.
+ * @param servers List of servers
+ * @param preferedScheme Optional. Prefered scheme to use
+ */
+function getPreferedServer(servers: ServerObject[], preferedScheme?: Schemes) {
```

Comment:
I think this code can be expressed in a more intuitive way. What do you think about the following? 

```
function getPreferredServer(servers: ServerObject[], preferredScheme?: Schemes) {
  if (Array.isArray(servers)) {
    if (servers.length === 0) {
      // No choice in servers can be made
      return servers[0]
    } else if (servers.length >= 1) {
      let preferredServers = servers.filter(server => server.url && server.url.startsWith(`${preferredScheme}:`))
        
      if (preferredServers.length > 0) {
        if (preferredServers.length > 1) {
          logHttp(`Warning: Randomly selected the first server that match the preferredScheme option (${preferredScheme}): ${preferredServer.url}`)
        }

        return preferredServers[0]
      } else {
        // No preferred scheme could be found
        logHttp(`Warning: Randomly selected first server ${preferredServer.url}`)
        return servers[0]
      }
    }
  }
  throw new Error('Cannot find a server to call')
}
```

Reply:
In the above code, `if (servers.length === 0) {` (line 3) is wrong. The if statement tries to cover the case when there is only a single server, in which case `server.length` should be `1`, though, not `0`.